tools/release_packages

```
@@ -0,0 +1,372 @@
+                    new_file.write(zulip_bots_line)
+                else:
+                    new_file.write(line)
+
+        os.remove(reqs)
+        shutil.move(temp_abs_path, reqs)
+
+    url_zulip = 'git+https://github.com/zulip/python-zulip-api.git@{tag}#egg={name}=={version}_git&subdirectory={name}\n'
+    url_zulip_bots = 'git+https://github.com/zulip/python-zulip-api.git@{tag}#egg={name}=={version}+git&subdirectory={name}\n'
+
+    _edit_reqs_file(
+        prod_lock,
+        url_zulip_bots.format(tag=hash_or_tag, name='zulip_bots', version=version),
+        url_zulip.format(tag=hash_or_tag, name='zulip', version=version),
+    )
+    _edit_reqs_file(
+        dev_lock,
+        url_zulip_bots.format(tag=hash_or_tag, name='zulip_bots', version=version),
+        url_zulip.format(tag=hash_or_tag, name='zulip', version=version),
+    )
+
+    editable_zulip = '-e "{}"\n'.format(url_zulip.rstrip())
+    editable_zulip_bots = '-e "{}"\n'.format(url_zulip_bots.rstrip())
+
+    _edit_reqs_file(
+        common,
+        editable_zulip_bots.format(tag=hash_or_tag, name='zulip_bots', version=version),
+        editable_zulip.format(tag=hash_or_tag, name='zulip', version=version),
+    )
+
+    message = 'Updated zulip API package requirements in the main repo.'
+    print(crayons.white(message, bold=True))
+
+    fh, temp_abs_path = tempfile.mkstemp()
+    with os.fdopen(fh, 'w') as new_file, open(version_py) as old_file:
+        for line in old_file:
+            if line.startswith('PROVISION_VERSION'):
+                version_num = float(line.split('=')[-1].strip().replace("'", ''))
+                version_num = version_num + 0.01
+                new_file.write("PROVISION_VERSION = '{}'\n".format(version_num))
+            else:
+                new_file.write(line)
+
+    os.remove(version_py)
+    shutil.move(temp_abs_path, version_py)
+
+    message = 'Incremented PROVISION_VERSION in the main repo.'
```

I'd make this method throw an exception if no `PROVISION_VERSION` line was found.

---

I don't think there is a scenario where `PROVISION_VERSION` won't exist. Shouldn't `zulip/version.py` always have it? What do you think? :)

---

I don't think it'll happen either, it's just that right now, for any reason we might or might not thinkg of, a missing `PROVISION_VERSION` would fail silently, and an exception is easy and quick to add here.

---

Cool, I agree, I'll do that then! :)